openmcyclus/DepleteReactor.py

```diff
@@ -0,0 +1,11 @@
+from cyclus.agents import Facility 
+
+class DepleteReactor(Facility):
+    '''
+    Sample facility to get things going
+    '''
+    def tick(self):
+        print("Hello ")
+
+    def tock(self):
+        print("World \n")
```

Comment:
Since we are at the beginning of this process, I think rather than placeholders we should think about what the design of this class is going to be and do our best to implement that using [abstract functions](https://docs.python.org/3/library/abc.html#abc.abstractmethod)

Reply:
Cyclus requires some pretty specific method names (see https://fuelcycle.org/arche/dre.html) to interact with the Dynamic Resource Exchange (DRE) and the different phases of a time step. If these aren't required, these are at least the convention I have seen in other python-based archetypes. `tick` and `tock` are specific method names to indicate things that should happen during the `Tick` and `Tock` phases of a time step. Right now I am working on writing some of these functions on the `functions` branch of my fork so that the `cycamore:Reactor` behavior is replicated in this archetype, before we add in functions to incorporate OpenMC. In the `depletion` branch in my fork I am working on functions to call the OpenMC `IndependentOperator` in a module separate from this class. I think that is where we will have the most discussion on design.

Reply:
I highly recommend you read through some of the pages on the website, so that we are all on a similar page about the Cyclus requirements.

Reply:
I see. I'll take a look at the docpages.